pkg/core/unspent_coin_state.go

```diff
@@ -122,7 +122,7 @@ func IsDoubleSpend(s storage.Store, tx *transaction.Transaction) bool {
 			if r.Err != nil {
 				return false
 			}
-			if unspent == nil {
+			if len(unspent.states) == 0 {
```

Comment:
This one is wrong, what would really match the C# code is something like
```go
if b, err := s.Get(key); err == nil {
...
} else {
    return true
}
```
But this case (`len(unspent.states) == 0`) also looks suspicious.

Reply:
it was matching this one:
```
if (states == null || ....
```
but anyway did like you mentioned

Reply:
It wasn't, that's the whole point. C#:
```c#
                UnspentCoinState state = persistence.UnspentCoins.TryGet(group.Key);
                if (state == null) return true;
```
In C# `TryGet` would return `null` in case there is no unspent coin state in the DB and in our case that's analogous to `s.Get()` returning an error instead of value for that key.

So now it's correct.